utils/s3/s3_retry_strategy.cc

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "s3_retry_strategy.hh"
+#include "aws_error.hh"
+#include "utils/log.hh"
+#include "utils/s3/creds.hh"
+
+using namespace std::chrono_literals;
+
+namespace aws {
+
+static logging::logger s3_retry_logger("s3_retry_strategy");
+
+s3_retry_strategy::s3_retry_strategy(seastar::timer<seastar::lowres_clock>& creds_update_timer, unsigned max_retries, unsigned scale_factor)
+    : default_retry_strategy(max_retries, scale_factor), _creds_update_timer(creds_update_timer) {
+}
+
+bool s3_retry_strategy::should_retry(const aws_error& error, unsigned attempted_retries) const {
+    if (attempted_retries < _max_retries && error.get_error_type() == aws_error_type::EXPIRED_TOKEN) {
+        s3_retry_logger.info("Credentials are expired, renewing");
+        _creds_update_timer.rearm(seastar::lowres_clock::now());
```

Comment:
Why is it enough? Below it returns `true` so client will wait for a "fixed" timeout, not until the token is really refreshed